### PR TITLE
Memory checker

### LINKED ISSUE #20: feat: Add a leak test
Create a leak test that:
- [x] launches valgrind with the needed options on `42sh`
- [x] executes some commands in it (why not use those of the functional tester)

Optionnal:
- [x] add it in the GitHub Action workflow
- [x] make the job fail if leaks/errors are detected

### PR DESCRIPTION
**Pull request by RezaRahemtola**
# Description

Leak tester to launch `valgrind` on the functional tests and see leaks and errors.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [ ] I have linked this PR to a milestone
- [ ] I have linked this PR to a project
- [x] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [x] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -119,6 +119,7 @@ debug_run:	fclean $(DEBUG_BINARY)
 tests_run:
 			$(MAKE) unit_tests
 			$(MAKE) func_tests
+			$(MAKE) mem_checks
```
Comment (RezaRahemtola) [inline Makefile:122]: make sure this won‘t display compilation logs as another PR is trying to make tests_run silently

File: Makefile [hunk 2]
```diff
@@ -127,5 +128,9 @@ func_tests:	re
 			@python3 -m pip install termcolor > /dev/null
 			python3 tests/tester.py -adc
 
+mem_checks:	fclean $(DEBUG_BINARY)
+			@python3 -m pip install termcolor > /dev/null
```
Comment (RezaRahemtola) [inline Makefile:137]: I already saw this line but I can‘t remember where...

File: tests/memory_checker.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+import argparse
+import enum
+import json
+import os
+import re
```
Comment (RezaRahemtola) [inline tests/memory_checker.py]: prefer `from module import function` when possible

**Review by RezaRahemtola — COMMENTED** [review]
otherwise LGTM

**Comment by Croos3r** [comment]
@Steci @EdenComp do you want to make me péter un boulon ? Please review

**Review by Steci — APPROVED** [review]
Good Tea 🍵

**Review by RezaRahemtola — COMMENTED** [review]
LGTM

**Pull request merged**